include/deal.II/base/work_stream.h

```diff
@@ -663,6 +663,163 @@ namespace WorkStream
 #  endif // DEAL_II_WITH_TBB
 
 
+
+#  ifdef DEAL_II_WITH_TASKFLOW
+    /**
+     * Mostly a copy of the 2nd implementation of the Workstream paper taking
+     * advantage of thread local lists for re-use. Uses taskflow for task
+     * scheduling rather than TBB. Currently does not support chunking.
+     */
+
+    namespace taskflow_no_coloring
+    {
+      template <typename Worker,
+                typename Copier,
+                typename Iterator,
+                typename ScratchData,
+                typename CopyData>
+
+      /**
+       *  The last two arguments in this function are for chunking support which
+       * currently does not exist but ideally will later. For now they are
+       * ignored but still here to permit existing programs to function
+       */
+      void
+      run(const Iterator                          &begin,
+          const typename identity<Iterator>::type &end,
+          Worker                                   worker,
+          Copier                                   copier,
+          const ScratchData                       &sample_scratch_data,
+          const CopyData                          &sample_copy_data,
+          const unsigned int queue_length = 2 * MultithreadInfo::n_threads(),
+          const unsigned int chunk_size   = 8)
+
+      {
+        struct ScratchDataObjects
+        {
+          std::unique_ptr<ScratchData> scratch_data;
+          bool                         currently_in_use;
+
+          /**
+           * Default constructor.
+           */
+          ScratchDataObjects()
+            : currently_in_use(false)
+          {}
+
+          ScratchDataObjects(std::unique_ptr<ScratchData> &&p,
+                             const bool                     in_use)
+            : scratch_data(std::move(p))
+            , currently_in_use(in_use)
+          {}
+
+          // Provide a copy constructor that actually doesn't copy the
+          // internal state. This makes handling ScratchAndCopyDataObjects
+          // easier to handle with STL containers.
+          ScratchDataObjects(const ScratchDataObjects &)
+            : currently_in_use(false)
+          {}
+        };
+
+        tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
+        tf::Taskflow  taskflow;
+
+        using ScratchDataList = std::list<ScratchDataObjects>;
+
+        Threads::ThreadLocalStorage<ScratchDataList> data;
+
+        tf::Task last_copier;
+
+        // Silence unused variable arguments.
+        (void)queue_length;
+        (void)chunk_size;
+
+        // This is used to connect each worker to its copier as communication
+        // between tasks is not supported.
+        unsigned int idx = 0;
```

Comment:
In general, starting a sentence with "This" is awkward because it's unclear what "this" refers to. Did you mean "The following variable..."? If so, however, I don't understand what the comment is saying. *How* are workers using this variable to make this connection?